hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java

```diff
@@ -229,13 +235,15 @@ private void moveServerRegionsFromGroup(Set<Address> servers, String targetGroup
           iter.remove();
         }
       }
+
+      retry++;
       try {
         rsGroupInfoManager.wait(1000);
       } catch (InterruptedException e) {
         LOG.warn("Sleep interrupted", e);
         Thread.currentThread().interrupt();
       }
-    } while (hasRegionsToMove);
+    } while (hasRegionsToMove && retry <= 50);
```

Comment:
For regions where move haven't succeed even after max number of retries, should we perform any cleanup (i.e. region state in meta and ZK)?

Reply:
As I have replied in jira, maybe cleanup is unsuitable.

Reply:
Agreed! Region move is attempted via TRSP, as you had pointed out in the jira, so whatever region state updates happen in ZK, meta, master cache, etc, should be handled at the TRSP level.